sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/VectorizedRleValuesReader.java

```diff
@@ -227,30 +230,340 @@ private void readBatchInternal(
         switch (mode) {
           case RLE:
```

Comment:
I am not too familiar with this code, but where do we assert that these are the only two encoding choices. There is no default arm in this case.

Reply:
The `MODE` enum is private to this class and the `mode` is initialized in this class, so I guess it should be fine. This maps to the Parquet spec for RLE/BitPacked encoding.